server/src/main/java/keywhiz/service/daos/ClientDAO.java

```diff
@@ -16,41 +16,62 @@
 
 package keywhiz.service.daos;
 
+import java.util.HashSet;
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
+import javax.inject.Inject;
 import keywhiz.api.model.Client;
-import org.skife.jdbi.v2.sqlobject.Bind;
-import org.skife.jdbi.v2.sqlobject.BindBean;
-import org.skife.jdbi.v2.sqlobject.GetGeneratedKeys;
-import org.skife.jdbi.v2.sqlobject.SqlQuery;
-import org.skife.jdbi.v2.sqlobject.SqlUpdate;
-import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper;
-import org.skife.jdbi.v2.sqlobject.customizers.SingleValueResult;
-
-@RegisterMapper(ClientMapper.class)
-public interface ClientDAO {
-  @GetGeneratedKeys
-  @SqlUpdate("INSERT INTO clients (name, createdBy, updatedBy, description, enabled, automationAllowed) " +
-      "VALUES (:name, :user, :user, :desc, true, false)")
-  public long createClient(@Bind("name") String name, @Bind("user") String user,
-      @Bind("desc") Optional<String> description);
-
-  @SqlUpdate("DELETE FROM clients WHERE id = :id")
-  public void deleteClient(@BindBean Client client);
-
-  @SingleValueResult(Client.class)
-  @SqlQuery("SELECT id, name, description, createdAt, createdBy, updatedAt, updatedBy, enabled, automationAllowed " +
-            "FROM clients WHERE name = :name")
-  public Optional<Client> getClient(@Bind("name") String name);
-
-  @SingleValueResult(Client.class)
-  @SqlQuery("SELECT id, name, description, createdAt, createdBy, updatedAt, updatedBy, enabled, automationAllowed " +
-      "FROM clients WHERE id = :id")
-  public Optional<Client> getClientById(@Bind("id") long id);
-
-  // Write update methods as needed.
-
-  @SqlQuery("SELECT id, name, description, createdAt, createdBy, updatedAt, updatedBy, enabled, automationAllowed " +
-            "FROM clients")
-  public Set<Client> getClients();
+import keywhiz.jooq.tables.records.ClientsRecord;
+import org.jooq.DSLContext;
+
+import static keywhiz.jooq.tables.Clients.CLIENTS;
+
+public class ClientDAO {
+  private final DSLContext dslContext;
+
+  @Inject
+  public ClientDAO(DSLContext dslContext) {
+    this.dslContext = dslContext;
+  }
+
+  public long createClient(String name, String user, Optional<String> description) {
+    ClientsRecord r = dslContext.newRecord(CLIENTS);
+
+    r.setName(name);
+    r.setCreatedby(user);
+    r.setUpdatedby(user);
+    r.setDescription(description.orElse(null));
+    r.setEnabled(true);
+    r.setAutomationallowed(false);
+    r.store();
+
+    return r.getId();
+  }
+
+  public void deleteClient(Client client) {
+    ClientsRecord r = dslContext.fetchOne(CLIENTS, CLIENTS.ID.eq(Math.toIntExact(client.getId())));
+    r.delete();
```

Comment:
If the client doesn't exist, I think this will throw NPE instead of a more descriptive sql-related error. Can you guard against that?